tools/sof-fw-boot-test.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+
+"""Module to stress test firmware boot"""
+
+import argparse
+import logging
+import sys
+from datetime import datetime
+from systemd import journal
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+# set path to the depbugfs entries
+DEBUGFS_PATH = "/sys/kernel/debug/sof/fw_debug_ops"
+
+def read_jctl_logs(start_time):
+    """Reads journalctl logs from a given start time and extracts the timestamp and log"""
+    journal_reader = journal.Reader()
+    journal_reader.this_boot()
+    journal_reader.seek_realtime(datetime.strptime(start_time, '%Y-%m-%d %H:%M:%S'))
+
+    entries = []
+    for entry in journal_reader:
+        timestamp = entry['__REALTIME_TIMESTAMP']
+        message = entry.get('MESSAGE', '')
+        if isinstance(message, bytes):
+            message = message.decode('utf-8', errors='replace')
+        entries.append((timestamp, message))
+
+    return entries
```

Comment:
In the future it would be good to make this an iterator with `yield` instead of loading everything in memory.